quadriga/src/main/java/edu/asu/diging/quadriga/web/EditCollectionController.java

```diff
@@ -0,0 +1,101 @@
+import org.springframework.ui.Model;
+import org.springframework.validation.BindingResult;
+import org.springframework.web.bind.annotation.RequestMapping;
+import org.springframework.web.bind.annotation.RequestMethod;
+import org.springframework.web.bind.annotation.RequestParam;
+import org.springframework.web.servlet.mvc.support.RedirectAttributes;
+
+import edu.asu.diging.quadriga.core.model.Collection;
+import edu.asu.diging.quadriga.core.service.CollectionManager;
+import edu.asu.diging.quadriga.web.forms.CollectionForm;
+
+/**
+ * This controller enables users to edit their collections
+ * 
+ * @author poojakulkarni
+ *
+ */
+@Controller
+public class EditCollectionController {
+
+    @Autowired
+    private CollectionManager collectionManager;
+
+    /**
+     * Request handler for getting the "Edit collections" view
+     * 
+     * @param id    of the collection to be edited
+     * @param model to set collectionForm attribute to pre-fill current name and
+     *              description
+     * @return the editCollection view
+     */
+    @RequestMapping(value = "/auth/collections/edit", method = RequestMethod.GET)
+    public String get(@RequestParam(value = "id", required = true) String id, Model model) {
+        Optional<Collection> collection = Optional.ofNullable(collectionManager.findCollection(id));
+
+        if (collection.isPresent()) {
+            Collection currentCollection = collection.get();
+            CollectionForm collectionForm = new CollectionForm();
+            collectionForm.setId(id);
+            collectionForm.setName(currentCollection.getName());
+            collectionForm.setDescription(currentCollection.getDescription());
+            model.addAttribute("collectionForm", collectionForm);
+
+            return "auth/editCollection";
+        }
+
+        return "redirect:/auth/collections";
+    }
+
+    /**
+     * Request handler for editing a collection in the "Edit collections" view
+     * 
+     * @param id                 of the collection to be edited
+     * @param collectionForm     that has the updated name and description values
+     * @param bindingResult      for checking form errors
+     * @param redirectAttributes used for adding flash attributes after redirecting
+     * @return
+     */
+    @RequestMapping(value = "/auth/collections/edit", method = RequestMethod.POST)
+    public String edit(@RequestParam(value = "id", required = true) String id, @Valid CollectionForm collectionForm,
+            BindingResult bindingResult, RedirectAttributes redirectAttributes) {
+        if (bindingResult.hasErrors()) {
+            redirectAttributes.addFlashAttribute("alert_type", "danger");
+            redirectAttributes.addFlashAttribute("alert_msg",
+                    "Something went wrong while editing collections, please try again!");
+            redirectAttributes.addFlashAttribute("show_alert", true);
+            return "auth/editCollection";
```

Comment:
the form just needs to show the validation results for the fields with the previously entered values still present.

Reply:
Didn't get you here?

The form does display pre-filled values while editing a collection and an alert in case the form is invalid.
The flash attributes here are added for displaying the result (success/failure) on UI after the editing is complete (i.e. after redirection to Show Collections page)